ksql-engine/src/test/resources/query-validation-tests/group-by.json

```diff
@@ -171,6 +171,46 @@
         {"topic": "OUTPUT", "key": "r0", "value": "4"}
       ]
     },
+    {
+      "name": "with aggregate arithmetic involving source field (stream->table)",
+      "statements": [
+        "CREATE STREAM TEST (ITEM INT, COST INT) WITH (kafka_topic='test_topic', value_format='DELIMITED');",
+        "CREATE TABLE OUTPUT AS SELECT ITEM, COST, COST * COUNT() FROM TEST GROUP BY ITEM, COST;"
+      ],
+      "inputs": [
+        {"topic": "test_topic", "value": "1,10"},
+        {"topic": "test_topic", "value": "1,20"},
+        {"topic": "test_topic", "value": "2,30"},
+        {"topic": "test_topic", "value": "1,10"}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": "1|+|10", "value": "1,10,10"},
+        {"topic": "OUTPUT", "key": "1|+|20", "value": "1,20,20"},
+        {"topic": "OUTPUT", "key": "2|+|30", "value": "2,30,30"},
+        {"topic": "OUTPUT", "key": "1|+|10", "value": "1,10,20"}
+      ]
+    },
+    {
+      "name": "with aggregate arithmetic involving source field (table->table)",
+      "statements": [
+        "CREATE TABLE TEST (f0 INT, f1 INT, f2 INT) WITH (kafka_topic='test_topic', key='f0', value_format='DELIMITED');",
+        "CREATE TABLE OUTPUT AS SELECT f1 * SUM(f2) FROM TEST GROUP BY f0;"
```

Comment:
Isn't this something we want to disallow? f1 can change from row to row, and the value used will just be whatever the last seen value was. Isn't this why we're restricting the usage of non-group-by columns in other select expressions?

Reply:
Absolutely - I've not updated since we discussed this.